api/transport/handler.go

```diff
@@ -168,13 +169,30 @@ func DispatchOnewayHandler(
 func DispatchStreamHandler(
 	h StreamHandler,
 	stream *ServerStream,
+	logger *zap.Logger,
 ) (err error) {
 	defer func() {
 		if r := recover(); r != nil {
-			log.Printf("Stream handler panicked: %v\n%s", r, debug.Stack())
-			err = fmt.Errorf("panic: %v", r)
+			err = logPanic("Stream", logger, r, stream.Request().Meta)
 		}
 	}()
 
 	return h.HandleStream(stream)
 }
+
+func logPanic(rpcType string, logger *zap.Logger, recovered interface{}, req *RequestMeta) error {
+	err := fmt.Errorf("panic: %v", recovered)
+	if logger != nil {
+		logger.Error(fmt.Sprintf("%s handler panicked", rpcType),
+			zap.String("service", req.Service),
+			zap.String("procedue", req.Procedure),
+			zap.String("encoding", string(req.Encoding)),
```

Comment:
should we merge from dev so we can also log req.Transport here?

Reply:
merge from dev if needed. the merge commit will get squashed away when this lands.

Reply:
another mantra worth remembering: [merge in, squash down]